.github/mergify.yml

```diff
@@ -62,6 +62,35 @@ pull_request_rules:
       - check-success=Bootstrap post job
       - 'check-success=docs/readthedocs.org:cabal'
 
+  # label when Mergify didn't trigger a merge automatically
+  - actions:
+      label:
+        add:
+          - waiting too long
+    name: Mergify hasn't merged a PR yet
+    conditions:
+      - base=master
+      - -draft
+      - -closed
+      - -merged
+      - or:
```

Comment:
This looks unnecessary? If we extend the set of such labels, we’ll have to update this place, so I’d rather avoid it. It should work just fine without it, right?

Reply:
As I read the Mergify documentation, `merged` and `closed` are separate states. They don't matter for the other ones because it would require a manual merge for there to be an issue, whereas here we are specifically checking for PRs that haven't been merged after `merge delay passed` and I don't want it to fire on every merged PR.

I admit I am guessing about the extra delay I gave it; it could quite possibly be shortened, but I don't want to experiment too much and have it fire on perfectly good PRs.

Reply:
I should add that GitHub also considers them separate (see the list of closed PRs), which is almost certainly why Mergify does.

Reply:
> merged and closed are separate states

Sorry for the confusion but I meant the `or` that checks the merge-me-style of labels.

Reply:
Mm, that might well be elidable given `merge delay passed`. Still being paranoid, but that one is probably unjustified.

Reply:
Although it does occur to me that we've had the merge label be removed without also removing `merge delay passed`, granting that nothing else checks for that possibility so it could lead to very unexpected merges.

Reply:
If we have an unexpected merge because of a label mishmash, we have a much bigger problem on our hands than a stuck PR. Putting the new label on such a PR hardly does any harm. Plus, these cases are relatively rare.

Reply:
Thanks for addressing it!